### ISSUE TITLE
Starling 2.0: Mask affects the color of rendered content 

### ISSUE THREAD

**Issue by joshtynjala** [Gamua/Starling-Framework#793]
I'm seeing a strange issue where a masked display object changes color. See the following screenshot. The Quad on the left is not masked, and the Quad on the right is masked by a red Quad.

![mask-color-issue](https://cloud.githubusercontent.com/assets/141885/12308153/73419ea4-b9f8-11e5-898a-87b413b57415.png)

Here's the code:

	var quad1:Quad = new Quad(100, 100, 0x999999);
	quad1.x = 20;
	quad1.y = 20;
	this.addChild(quad1);

	var quad2:Quad = new Quad(100, 100, 0x999999);
	quad2.x = 140;
	quad2.y = 20;
	this.addChild(quad2);
	
	var mask:Quad = new Quad(100, 100, 0xff0000);
	quad2.mask = mask;

Interestingly, if you change the color of the mask quad, the discoloration changes to match. It's like the mask is rendering at a very small alpha value that is just slightly larger than 0.

**Comment by PrimaryFeather** [comment]
Wow, that's odd ...! Thanks for the heads-up, I'll take a look at that!

**Comment by PrimaryFeather** [comment]
The problem was in the VertexData class: with premultiplied alpha, any zero alpha value will make all RGB data disappear (it's multiplied by zero). Thus, in Starling 1.x, I simply introduced a minimal alpha value of 0.001. This worked, because all colors were stored as "double", and in the 8 bit per channel rendering accuracy, it would still equal a zero alpha value.

In the new version, however, color data is *stored* with 8 bit per channel. Thus, I thought I'd simply raise the minimum alpha value to 0.02 (5/255) instead. But this is — as you discovered — visible in the output. Furthermore, after thinking about it a little longer, I don't think it makes much sense. Yes, it will prevent the RGB data to be lost altogether; but we'll still end up with only 5 different values per channel — which is not much better than losing it completely.

Thus, I removed that minimal value altogether. People just have to beware that PMA will affect the accuracy of the RGB channels.

Thanks for making me aware of that! :smile:
